pkgs/tools/typesetting/tex/texpresso/tectonic.nix

```diff
@@ -16,18 +10,23 @@ tectonic-unwrapped.override (old: {
           };
           useFetchCargoVendor = true;
           cargoHash = "sha256-mqhbIv5r/5EDRDfP2BymXv9se2NCKxzRGqNqwqbD9A0=";
+          # rebuild cargoDeps by hand because `.overrideAttrs cargoHash`
+          # does not reconstruct cargoDeps (a known limitation):
+          cargoDeps = rustPlatform.fetchCargoVendor {
+            inherit (finalAttrs) src;
+            name = "${finalAttrs.pname}-${finalAttrs.version}";
+            hash = finalAttrs.cargoHash;
+            patches = finalAttrs.cargoPatches;
+          };
```

Comment:
Well done! I was very close but my internet connection was too slow to try out things in a reasonable manner :).